provider/kubernetes.go

```diff
@@ -145,29 +153,22 @@ func (provider *Kubernetes) loadIngresses(k8sClient k8s.Client) (*types.Configur
 						}
 					}
 				}
-				if len(pa.Path) > 0 {
-					ruleType := i.Annotations["traefik.frontend.rule.type"]
 
-					switch strings.ToLower(ruleType) {
-					case "pathprefixstrip":
-						ruleType = "PathPrefixStrip"
-					case "pathstrip":
-						ruleType = "PathStrip"
-					case "path":
-						ruleType = "Path"
-					case "pathprefix":
-						ruleType = "PathPrefix"
-					case "":
-						ruleType = "PathPrefix"
-					default:
-						log.Warnf("Unknown RuleType %s for %s/%s, falling back to PathPrefix", ruleType, i.ObjectMeta.Namespace, i.ObjectMeta.Name)
-						ruleType = "PathPrefix"
+				if len(pa.Path) > 0 {
+					ruleType, unknown := getRuleTypeFromAnnotation(i.Annotations)
+					switch {
```

Comment:
This may be a dumb question, but If you are not switching on a variable, should you still have a default in the event that neither of the cases are matched?